wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -69,6 +71,39 @@ protected final Object writeReplace() throws ObjectStreamException {
     return new MessageSerializedForm<>(this, (Class<Message>) getClass());
   }
 
+  /** Returns a new builder for this class, initialized with the data in this message. */
+  public final <B extends Builder<T, B>> B newBuilder() {
```

Comment:
We could eliminate the builder's copy constructor and replace it with an abstract instance method here which was implemented on each message as the contents of the copy constructor.